### ISSUE TITLE
"Could not resolve a module for a custom reporter" error, tests don't run

### ISSUE THREAD

**Issue by JJHDoubleD** [jest-community/vscode-jest#885]
### Environment

1. `vscode-jest version`: 4.6.0
2. `node -v`: 18.4
3. `npm -v` or `yarn --version`: 1.22.19
4. `npm ls jest` or `npm ls react-scripts` (if you haven’t ejected): jest@27.5.1
5. your vscode-jest settings if customized:
    - jest.jestCommandLine? `yarn test:local:ext`
    - jest.autoRun? {}
    - anything else that you think might be relevant?  **This doesn't happen if I don't run it through Yarn, e.g. by using
 `"jest.jestCommandLine": "npx jest --config ./test/jest-local.js"`
 But then I run into the NPM escaping issue (https://github.com/jest-community/vscode-jest/issues/838#issuecomment-1051323196) instead so I still can't run my tests.**
6. Operating system: Win11

`"test:local:ext": "jest --config ./test/jest-local.js"` works fine when run from CLI.

### Prerequisite
- are you able to run jest test from the command line? Yes
- how do you run your tests from the command line? (for example: `npm run test` or `node_modules/.bin/jest`) `yarn test:local:ext`

### Steps to Reproduce
jest-local.js
```
module.exports = {
  moduleFileExtensions: ['js', 'json', 'ts'],
  rootDir: '..',
  testEnvironment: 'node',
  testRegex: '.spec.ts$',
  transform: {
    '^.+\\.(t|j)s$': 'ts-jest',
  },
  moduleNameMapper: {
    '^@config/(.*)$': '<rootDir>/src/config/$1',
    '^@common/(.*)$': '<rootDir>/src/common/$1',
    '^@module/(.*)$': '<rootDir>/src/module/$1',
    '^@provider/(.*)$': '<rootDir>/src/provider/$1',
  },
  setupFiles: ['<rootDir>/test/jest.local.env.ts'],
};
```

### Relevant Debug Info

```
[project-name]: executing profile: "run" for c:\Users\JJH\Developer\project-name\src\module\auth\auth.service.spec.ts...
yarn run v1.22.19
$ jest --config ./test/jest-local.js --testLocationInResults --json --useStderr --outputFile C:\Users\JJH\AppData\Local\Temp\jest_runner_dp_account_2.json --no-coverage --reporters default --reporters c:\Users\JJH\.vscode\extensions\orta.vscode-jest-4.6.0\out\reporter.js --colors --watchAll=false --testPathPattern c:\\Users\\JJH\\Developer\\project-name\\src\\module\\auth\\auth\.service\.spec\.ts
Error: Could not resolve a module for a custom reporter.
  Module name: c:UsersJJH.vscodeextensionsorta.vscode-jest-4.6.0outreporter.js
    at C:\Users\JJH\Developer\project-name\node_modules\jest-config\build\normalize.js:558:15
    at Array.map (<anonymous>)
    at normalizeReporters (C:\Users\JJH\Developer\project-name\node_modules\jest-config\build\normalize.js:540:33)
    at normalize (C:\Users\JJH\Developer\project-name\node_modules\jest-config\build\normalize.js:699:5)
    at readConfig (C:\Users\JJH\Developer\project-name\node_modules\jest-config\build\index.js:236:74)
    at async readConfigs (C:\Users\JJH\Developer\project-name\node_modules\jest-config\build\index.js:420:26)
    at async runCLI (C:\Users\JJH\Developer\project-name\node_modules\@jest\core\build\cli\index.js:132:59)
    at async Object.run (C:\Users\JJH\Developer\project-name\node_modules\jest-cli\build\cli\index.js:155:37)
```

**Comment by connectdotz** [comment]
hi @JJHDoubleD can you try the jest command line generated above directly in the terminal (you might need to give an explicit jest path):
```
jest --config ./test/jest-local.js --testLocationInResults --json --useStderr --outputFile C:\Users\JJH\AppData\Local\Temp\jest_runner_dp_account_2.json --no-coverage --reporters default --reporters c:\Users\JJH\.vscode\extensions\orta.vscode-jest-4.6.0\out\reporter.js --colors --watchAll=false --testPathPattern c:\\Users\\JJH\\Developer\\project-name\\src\\module\\auth\\auth\.service\.spec\.ts
```
Assuming it does generate the same error, will need you to try a few things about the problem reporter path ("c:\Users\JJH\.vscode\extensions\orta.vscode-jest-4.6.0\out\reporter.js") to help us diagnose the issue: 
First, make sure this file does exist in your system. 

1. try to use the upperCase drive letter 'C'
2. try to double escape the `\` 
3. try using the forward-slash `/`  instead of backslash

let me know if any of them made any difference. 

I am somewhat puzzled that if it was indeed the reporter path issue, we should have seen a lot more people reporting this issue in Windows but that did not seem to be the case... If you have a sample repo that will greatly help speed up the diagnosis.

**Comment by connectdotz** [comment]
ok, looks like you found something working. 👍 

If you are still interested in trying npm, they claimed to fix the related issue npm/cli#5004. I didn't get a chance to try it yet. If you got a few cycles, please feel free to try the latest npm version (8.15 or 8.14) and let us know if the `testPathPattern` is still messed up. Thanks.

**Comment by connectdotz** [comment]
for people who had problems with testPathPattern, i.e. npm miss-interpreting the pass-through arguments: it seems npm @>=8.15 has fixed this issue. 🎉
